CI unit test failures

Recently, CI unit test workflows have started to fail with reports such as:

```py
=========================== short test summary info ============================
FAILED apstools/plans/tests/test_alignment.py::test_lineup2[parms4] - ophydregistry.exceptions.MultipleComponentsFound: "Found 2 components matching query [any_of=None, label=None, name=noisy]. Consider using ``findall()``. [EpicsSignalRO(read_pv='gp:userCalc1.VAL', name='noisy', value=925.7771208487127, timestamp=1744889325.064775, auto_monitor=True, string=False), EpicsSignalRO(read_pv='gp:userCalc10.VAL', name='noisy', value=55.0, timestamp=1744889036.278962, auto_monitor=True, string=False)]"
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
=========== 1 failed, 492 passed, 303 warnings in 644.54s (0:10:44) ============
```

----

Blocks these PRs:

- #1063
- #1065
- #1066

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -42,6 +42,7 @@ describe future plans.
    * Now testing with Python versions 3.9 - Py3.12.
      (Can't use with Py3.13 yet due to databroker 1.2.5 pin.)
    * Removed 'lifetime' component from 'aps_machine' device.
+   * Refactor unit test for change in upstream ophydregistry findall().
 
 1.7.2
 *****

--- a/apstools/plans/tests/_scaler.py
+++ b/apstools/plans/tests/_scaler.py
@@ -40,11 +40,11 @@ class OverrideSourceRO(ophyd.EpicsSignalRO):
 
     def get(self, **kwargs):
         """."""
-        signal = oregistry.find(
-            name=self.parent.override_signal_name.get(),
-            allow_none=True,
-        )
-        return (signal or super()).get()
+        pvname = self.parent.override_signal_name.get()
+        signals = oregistry.findall(pvname, allow_none=True)
+        # Multiple signals for identical PV name are ok here.
+        signal = super() if len(signals) == 0 else signals[0]
+        return signal.get()
 
 
 class ScalerChannel(ophyd.scaler.ScalerChannel):

--- a/docs/source/_static/switcher.json
+++ b/docs/source/_static/switcher.json
@@ -5,7 +5,12 @@
         "url": "https://bcda-aps.github.io/apstools/dev/"
     },
     {
-        "name": "1.7.2 (latest)",
+        "name": "1.7.3 (latest)",
+        "version": "1.7.3",
+        "url": "https://bcda-aps.github.io/apstools/1.7.3/"
+    },
+    {
+        "name": "1.7.2",
         "version": "1.7.2",
         "url": "https://bcda-aps.github.io/apstools/1.7.2/"
     },

--- a/environment.yml
+++ b/environment.yml
@@ -1,48 +1,49 @@
 name: apstools-test-environment
 
 channels:
-  - apsu  # First, for compatible epics-base and aps-dm-api
-  - conda-forge
-  - nodefaults
+- apsu # First, for compatible epics-base and aps-dm-api
+- conda-forge
+- nodefaults
 
 dependencies:
-  - python >=3.9
-  - area-detector-handlers
-  - bluesky >=1.6.7, !=1.11.0
-  - bluesky-live
-  - databroker =1.2
-  - databroker-pack
-  - entrypoints
-  - h5py
-  - intake <=1
-  - matplotlib-base
-  - mongoquery >=1.3.6
-  - numpy <2
-  - openpyxl
-  - ophyd >=1.6.1
-  - pandas
-  - pint
-  - pip
-  - prompt-toolkit
-  - psutil
-  - pydata-sphinx-theme
-  - pyEpics >=3.4.3
-  - pymongo
-  - pyqt =5
-  - pyRestTable >=2020.0.8
-  - pysumreg
-  - qt =5
-  - readline !=8.1.2
-  - scipy
-  - setuptools-scm
-  - spec2nexus
-  - sphinx >=5
-  - toolz
+- python >=3.9
+- area-detector-handlers
+- bluesky >=1.6.7, !=1.11.0
+- bluesky-live
+- databroker =1.2
+- databroker-pack
+- entrypoints
+- h5py
+- intake <=1
+- matplotlib-base
+- mongoquery >=1.3.6
+- numpy <2
+- openpyxl
+- ophyd >=1.6.1
+- pandas
+- pandoc
+- pint
+- pip
+- prompt-toolkit
+- psutil
+- pydata-sphinx-theme
+- pyEpics >=3.4.3
+- pymongo
+- pyqt =5
+- pyRestTable >=2020.0.8
+- pysumreg
+- qt =5
+- readline !=8.1.2
+- scipy
+- setuptools-scm
+- spec2nexus
+- sphinx >=5
+- toolz
 
-  - aps-dm-api >=8   # linux-64  osx-64
+- aps-dm-api >=8 # linux-64  osx-64
 
-  - pip:
-    # - bluesky_live
-    # - ipython-genutils==0.2.0
-    # - prompt-toolkit==1.0.15
-    - ophyd-registry
+- pip:
+  # - bluesky_live
+  # - ipython-genutils==0.2.0
+  # - prompt-toolkit==1.0.15
+  - ophyd-registry

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -74,6 +74,39 @@ dependencies = [
   "xlrd",
 ]
 
+[project.optional-dependencies]
+dev = [
+  "build",
+  "coverage",
+  "coveralls",
+  "isort",
+  "mypy",
+  "pre-commit",
+  "pytest",
+  "ruff",
+  # databroker 2.0.0b57 needs
+  "doct",
+  "humanize",
+  "mongomock",
+  "pymongo",
+  # tiled 0.1.0b19 needs
+  # "suitcase-mongo",
+]
+doc = [
+  "ipython",
+  "jinja2 >=3.0",
+  "markupsafe ==2.0.1",
+  "myst-parser",
+  "nbsphinx",
+  "pydata-sphinx-theme",
+  "pygments >=2.12",
+  "pygments-ipython-console",
+  "setuptools-scm",
+  "sphinx",
+  "sphinx-design",
+]
+all = ["apstools[dev,doc]"]
+
 [project.scripts]
 spec2ophyd = "apstools.migration.spec2ophyd:main"
 

